express.js

```diff
@@ -322,7 +322,12 @@ setInterval(() => {
 app.use(function (req, res) {
   res.status(404).render('errors/404');
 });
-app.use(function (req, res) {
+app.use(function (err, req, res, next) {
+  console.error('[error] Incoming request to"', req.originalUrl, '"failed with error "', err.toString(), '"')
+  if (res.headersSent) {
+    return next(err);
+  }
```

Comment:
What does this do?

Reply:
there is an issue if the error is thrown after the server already sent the http-header, in this case it's better to let express handle the error, see https://expressjs.com/en/guide/error-handling.html
There are some pending issues with the general approach.. how to handle errors if the caller expects json, js etc.

Reply:
if you missed it, this function was never called (thus no error handling on production) bc. of express expecting 4 parameters:
`Define error-handling middleware functions in the same way as other middleware functions, except error-handling functions have four arguments instead of three: (err, req, res, next)`